presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -585,5 +596,119 @@ public void prepareFinal()
                 accumulator.addInput(groupIds, page);
             });
         }
+
+        @Override
+        public void addIntermediateStateOnUnspill(GroupByIdBlock groupIds, Block[] blocks)
+        {
+            Block[] groupByAndBlocks = new Block[blocks.length + 1];
+            System.arraycopy(blocks, 0, groupByAndBlocks, 0, blocks.length);
```

Comment:
Not too happy about having to do this since unsure of the memory implications. But this is the only way since the two accumulators format their pages in different ways. Let me know if you can think of any alternatives.